cmd/soroban-cli/src/commands/network/start.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::HashMap;
+
+use bollard::{
+    container::{Config, CreateContainerOptions, StartContainerOptions},
+    image::CreateImageOptions,
+    service::{HostConfig, PortBinding},
+};
+use futures_util::TryStreamExt;
+
+use crate::commands::network::shared::{connect_to_docker, Network, DOCKER_SOCKET_PATH_HELP};
+
+const DEFAULT_PORT_MAPPING: &str = "8000:8000";
+const DOCKER_IMAGE: &str = "docker.io/stellar/quickstart";
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("⛔ ️Failed to start container: {0}")]
+    StartContainerError(#[from] bollard::errors::Error),
+}
+
+#[derive(Debug, clap::Parser, Clone)]
+pub struct Cmd {
+    /// Network to start
+    pub network: Network,
+
+    #[arg(short = 'd', long, help = DOCKER_SOCKET_PATH_HELP)]
```

Comment:
This is the official ENV for docker tooling to check. It lets it be flexible to both sockets and http endpoints.
```suggestion
    #[arg(short = 'd', long, help = DOCKER_SOCKET_PATH_HELP, env = "DOCKER_HOST")]
```

Reply:
nice!!